### PR TITLE
[terra-clinical-item-collection][terra-clinical-item-display] Accessibility failures

### LINKED ISSUE #643: [terra-clinical-item-collection][terra-clinical-item-display] Accessibilities errors reported by axe-core
# Bug Report
terra-clinical-item-collection and terra-clinical-item-display have accessibility failures in wdio tests when running agains axe-core 3.5.0

## Description
- terra-clinical-item-collection - A selectable List with the `listbox` role requires an `aria-label`

- terra-clinical-item-display - A disabled item display has an `opacity` style of `0.2` and this causes insufficient color-contrast. `aria-disabled` needs to be set for a disabled item display.

## Steps to Reproduce
1. Pull in axe-core 3.5.0 and run the wdio tests to see the failures

## Additional Context / Screenshots

> expected no accessibility violations but got:
> 	{
>   "description": "Ensures every ARIA input field has an accessible name",
>   "help": "ARIA input fields have an accessible name",
>   "helpUrl": "https://dequeuniversity.com/rules/axe/3.5/aria-input-field-name?application=axeAPI",
>   "id": "aria-input-field-name",
>   "impact": "serious",
>   "nodes": [
>     {
>       "all": [],
>       "any": [
>         {
>           "data": null,
>           "id": "aria-label",
>           "impact": "serious",
>           "message": "aria-label attribute does not exist or is empty",
>           "relatedNodes": []
>         },
>         {
>           "data": null,
>           "id": "aria-labelledby",
>           "impact": "serious",
>           "message": "aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty",
>           "relatedNodes": []
>         },
>         {
>           "data": null,
>           "id": "non-empty-title",
>           "impact": "serious",
>           "message": "Element has no title attribute or the title attribute is empty",
>           "relatedNodes": []
>         }
>       ],
>       "failureSummary": "Fix any of the following:\n  aria-label attribute does not exist or is empty\n  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty\n  Element has no title attribute or the title attribute is empty",
>       "html": "<ul data-terra-clinical-item-collection-list-view=\"true\" role=\"listbox\" class=\"List-module__list___4A7Nh\">",
>       "impact": "serious",
>       "none": [],
>       "target": [
>         "ul"
>       ]
>     }
>   ],
>   "tags": [
>     "wcag2a",
>     "wcag412"
>   ]
> }
> 

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by benbcai**
### Summary
Fix accessibility failures in wdio tests when running agains axe-core 3.5.3

Closes #643

### Deployment Link

https://terra-clinical-deployed-pr-644.herokuapp.com/

### Testing
Updated wdio and jest tests.

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-item-collection/src/ItemCollection.jsx [hunk 1]
```diff
@@ -93,6 +93,7 @@ const ItemCollection = (props) => {
     ...customProps
   } = props;
 
+  const ariaLabel = customProps['aria-label'];
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/src/ItemCollection.jsx:96]: Should we make this a first-class prop?

Comment (emilyrohrbough) [reply packages/terra-clinical-item-collection/src/ItemCollection.jsx:96]: this is spreading on the div wrapping the responsive element below. Should it be removed? Should it spread on the table?

File: packages/terra-clinical-item-collection/tests/jest/__snapshots__/ItemCollection.test.jsx.snap [hunk 2]
```diff
@@ -445,12 +445,14 @@ exports[`should render an ItemCollection with an end accessory 1`] = `
 
 exports[`should render an ItemCollection with custom props 1`] = `
 <div
+  aria-label="Item Collection"
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/tests/jest/__snapshots__/ItemCollection.test.jsx.snap:448]: is this correct?

File: packages/terra-clinical-item-collection/src/ItemCollection.jsx (hunk 1 above)
Comment (benbcai) [reply packages/terra-clinical-item-collection/src/ItemCollection.jsx]: Added ariaLabel as a first-class prop. The label is no longer being spread on the wrapping div and also passed down to table. 
https://github.com/cerner/terra-clinical/pull/644/commits/75438d43ea6e9d600bde3846d966c3082372e338

File: packages/terra-clinical-item-collection/tests/jest/__snapshots__/ItemCollection.test.jsx.snap (hunk 2 above)
Comment (benbcai) [reply packages/terra-clinical-item-collection/tests/jest/__snapshots__/ItemCollection.test.jsx.snap]: The label is no longer applied to the div. https://github.com/cerner/terra-clinical/pull/644/commits/75438d43ea6e9d600bde3846d966c3082372e338

**Pull request merged**